### PR TITLE
Feature/Upgrade Markdown

### PR DESCRIPTION
**Pull request by VirginiaDooley**
This work closes https://github.com/DemocracyClub/Website/issues/390 and https://github.com/DemocracyClub/Website/issues/388 and includes
- [x] Removing Markdown-deux
- [x] Writing custom template tag and use that to render markdown
- [x] Updating the new report system
- [x] Changing all report html templates to md files and update apps and/or urls accordingly
- [x] Increasing test coverage
Screen Shot 2021-10-05 at 1 17 06 PM

### CODE REVIEW

File: democracy_club/apps/core/templatetags/markdown_filter.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import markdown
+
+register = template.Library()
+
+
+@register.filter(name="markdown")
+def markdown_filter(value, style="default"):
+    try:
+        return mark_safe(markdown.markdown(value, style))
+    except ImportError:
```
Comment (symroe) [inline democracy_club/apps/core/templatetags/markdown_filter.py:19]: This exception will never be raised here, as we're importing Markdown further up in this file. The way to get this working would be to add the `import markdown` line inside the `try` block here.

To be honest though I don't think it's a very useful exception for us to have — without markdown installed we'll get the same sort of exception anyway, so we may as well remove the try/except here

File: democracy_club/apps/core/templatetags/markdown_filter.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+
+class MarkdownNode(template.Node):
+    def __init__(self, style, nodelist):
+        self.style = style
+        self.nodelist = nodelist
+
+    def render(self, context):
+        value = self.nodelist.render(context)
+        try:
```
Comment (symroe) [inline democracy_club/apps/core/templatetags/markdown_filter.py:54]: And here

File: democracy_club/apps/core/report_helpers/views.py [hunk 3]
```diff
@@ -19,9 +19,7 @@ def get_context_data(self, **kwargs):
         template = Template(self.markdown_content())
         rendered_template = template.render(RequestContext(self.request))
 
-        markdown_doc = markdown_deux.markdown(
-            rendered_template, style="default"
-        )
+        markdown_doc = markdown.markdown(rendered_template, style="default")
```
Comment (symroe) [inline democracy_club/apps/core/report_helpers/views.py:22]: Here's the docs on how to add TOC using this version of Markdown: 

https://python-markdown.github.io/extensions/toc/

Comment (symroe) [reply democracy_club/apps/core/report_helpers/views.py:22]: This means we can get rid of `context["toc"]` and the related line in the template file

File: democracy_club/apps/projects/templates/projects/cvs.html [hunk 4]
```diff
@@ -1,30 +1,29 @@
...
-Democracy Club CVs
+    Democracy Club CVs
 {% endblock %}
 
 
 {% block content %}
-<div class="ds-stack-smaller">
-    <h1>Democracy Club CVs</h1>
-{% filter markdown %}
+    <div class="ds-stack-smaller">
```
Comment (symroe) [inline democracy_club/apps/projects/templates/projects/cvs.html]: Are all these indents ok? Has that issue gone away with the new filter?

Comment (VirginiaDooley) [reply democracy_club/apps/projects/templates/projects/cvs.html]: No. Still indenting.

File: democracy_club/apps/core/tests/test_markdown.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
+from markdown.test_tools import TestCase
```
Comment (VirginiaDooley) [inline democracy_club/apps/core/tests/test_markdown.py:1]: I'm not sure how useful this is - I could obviously take this much further. We are already testing valid html elsewhere.

**Comment by VirginiaDooley** [comment]
If I understand the coveralls report correctly, I'm not sure the remaining drop in coverage is related to this work (seems donations needs a boost, happy to do that in another PR).

File: democracy_club/apps/core/tests/test_markdown.py (hunk 5 above)
Comment (VirginiaDooley) [reply democracy_club/apps/core/tests/test_markdown.py:1]: I added additional smoke tests but want I want to do is actually pass a markdown file as an arguement. However, I'm struggling to find examples.

**Comment by VirginiaDooley** [comment]
I tried to deploy this to staging and got this error: 
`Warning! Couldn't get function democracy-club-dev in eu-west-1 - have you deployed yet?`
I used `aws configure` to enter the credentials but the same error continues to occur after `./deploy.sh dev`

**Pull request merged**